pkg/blockdeviceclaim/v1alpha1/build.go

```diff
@@ -108,8 +108,27 @@ func (b *Builder) WithAnnotations(annotations map[string]string) *Builder {
 	return b
 }
 
-// WithLabels sets the Labels field of PVC with provided arguments
-func (b *Builder) WithLabels(labels map[string]string) *Builder {
+// WithAnnotations appends or overwrites existing Annotations
+// values of BDC with provided arguments
+func (b *Builder) WithAnnotations(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build BDC object: missing annotations"),
+		)
+		return b
+	}
+	if b.BDC.Object.Annotations == nil {
+		return b.WithAnnotationsNew(annotations)
```

Comment:
:+1: for calling WithAnnotationsNew here.. but, for some reason, I don't see calling WithLabelsNew from WithLabels for same case